frontend-react/src/content/resources/api-programmers-guide/sidenav.mdx

```diff
@@ -0,0 +1,40 @@
+import { SideNav } from "@trussworks/react-uswds"
+import { USNavLink } from "../../../components/USLink"
+import { SideNavItem } from "../../../components/SideNav"
+
+<SideNav items={[
```

Comment:
I realize this pull request is still in draft, but I wonder if we could boil this down into an array that'd be a little easier to maintain/more resistant to typos if we need to update in the future:

```ts
const reportStreamApiNav = {
    path: "/reportstream-api",
    title: "ReportStream API",
    items: [
        {
            path: "#onboarding-overview",
            title: "Onboarding overview",
        },
        ...,
    ],
};

const gettingStartedNav = {
    path: "/getting-started",
    title: "Getting started",
    items: [...]
};

const documentationNav = {
    path: "/documentation",
    title: "Documentation",
    isActive: true,
    items: [...]
};

const navigationItems = [
    {
        path: "/resources",
        title: "",
        items: [
            reportStreamApiNav,
            gettingStartedNav,
            documentationNav
        ],
    },
];
```

Then we could recurse through each `items` value and then render a SideNavItem if there are nested items or a USNavLink if there aren't.  Might be overengineering, but I figured it was worth bringing up!

Reply:
The thought crossed my mind but initial instinct was to not add more complexity for initial implementation, but if breaking this down into simpler data types is OK then it shall be done.

Reply:
It's okay to do something like this in my mind, but it's not a blocker either way!  I'd say maybe spent at most at most like ten-fifteen minutes on this if you want and move on if it feels like it's getting unnecessarily complex.